gemrb/plugins/FXOpcodes/FXOpcodes.cpp

```diff
@@ -4661,13 +4661,13 @@ int fx_cast_spell (Scriptable* Owner, Actor* target, Effect* fx)
 		// no deplete, no interrupt, caster or provided level
 		// ForceSpell doesn't have a RES variant, so there's more work
 		std::string tmp;
+		tmp = fmt::format("ForceSpell({},{})", target == Owner ? "Myself" : "[-1]", ResolveSpellNumber(fx->Resource));
+		Action *forceSpellAction;
 		if (target == Owner) {
-			// charname has no scriptname ...
-			tmp = fmt::format("ForceSpell(Myself,{})", ResolveSpellNumber(fx->Resource));
+			forceSpellAction = GenerateAction(std::move(tmp));
 		} else {
-			tmp = fmt::format("ForceSpell(\"{}\",{})", target->GetScriptName(), ResolveSpellNumber(fx->Resource));
+			forceSpellAction = GenerateActionDirect(std::move(tmp), target);
```

Comment:
This will work for the Myself case as well, so the ternary and branching can be removed.

Reply:
Oh, all right.  I assumed there was a reason to special-case it.